irods_specific_implementation/irods_rest_meta_api_implementation.h

```diff
@@ -0,0 +1,161 @@
+#ifndef IRODS_REST_CPP_QUERY_API_IMPLEMENTATION_H
+#define IRODS_REST_CPP_QUERY_API_IMPLEMENTATION_H
+
+#include "irods_rest_api_base.h"
+
+#include "constants.hpp"
+#include <irods/metadata.hpp>
+
+#include <pistache/router.h>
+#include <nlohmann/json.hpp>
+
+
+#define OP_STATUS_SUCCESS "SUCCESS"
+#define OP_STATUS_FAILED "FAILURE"
```

Comment:
Prefer const variables over macros.

Macros cannot be tracked by the debugger. These could be moved into the metadata class as private static const variables given that they are only used by this endpoint.